### PR TITLE
Fix: Local Client Factory was not properly handling the override for auth

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

**Review by dennisreimann — APPROVED** [review]
LGTM. For context: We found this bug while working on btcpayserver/btcpayserver#3790.

File: BTCPayServer/Controllers/GreenField/LocalBTCPayServerClient.cs [hunk 1]
```diff
@@ -303,62 +303,80 @@ private readonly GreenfieldStoreLightningNetworkPaymentMethodsController
...
             }
 
-            public Task<IEnumerable<IAuthorizationHandler>> GetHandlersAsync(AuthorizationHandlerContext context)
+            public Task<AuthorizationResult> AuthorizeAsync(ClaimsPrincipal user, object resource, string policyName)
             {
-                return Task.FromResult<IEnumerable<IAuthorizationHandler>>(new IAuthorizationHandler[]
-                {
-                    new LocalGreenfieldAuthorizationHandler(_httpContextAccessor, _userManager, _storeRepository)
-                });
+                throw new  NotImplementedException();
```
Comment (fernandohre) [inline BTCPayServer/Controllers/GreenField/LocalBTCPayServerClient.cs:379]: Avoid throwing this exception or change back to what was doing before.

Comment (Kukks) [reply BTCPayServer/Controllers/GreenField/LocalBTCPayServerClient.cs]: This was a completely different interface implementation and was a method that supposedly would never be used. Either way, I implemented it now.

**Comment by pavlenex** [comment]
@fernandohre Thank you for your recent reviews on PR's, not something people contribute with, so much appreciated, and please if you have the capacity, keep reviewing 💪

**Comment by NicolasDorier** [comment]
please don't change formatting this is making things painful to review :(

**Comment by fernandohre** [comment]
> @fernandohre Thank you for your recent reviews on PR's, not something people contribute with, so much appreciated, and please if you have the capacity, keep reviewing 💪
> 
> If you'd like to engage with our community ,feel free to join our Mattermost (download mattermost app) and join chat.btcpayserver.org if you haven't already 🚀

I didn't know, I'll be there then. Thank you I'll continue

**Pull request merged**